### ISSUE TITLE
MembersPage: strange FormatException

### ISSUE THREAD

**Issue by bartekpacia** [bartekpacia/mentorship-client#36]
**Describe the bug**
I just opened `MembersPage` and this showed up

**To Reproduce**
1. Go to `MembersPage`
2. See strange error/don't see users 

**Expected behavior**
It should display a list of all verified users.

**Screenshots**
<img width="334" alt="Screenshot 2020-05-03 at 2 44 16 AM" src="https://user-images.githubusercontent.com/40357511/80895748-95b21a00-8ce8-11ea-9262-abc59971f0c1.png">

**Logs**
```
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.600469: MembersPageSelected
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.605046: Transition { currentState: HomeScreenStats, event: MembersPageSelected, nextState: HomeScreenMembers }
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.788853: ProfilePageShowed
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.825160: RelationPageShowed
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.883083: Transition { currentState: ProfilePageInitial, event: ProfilePageShowed, nextState: ProfilePageLoading }
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.893891: --> GET https://mentorship-backend-temp.herokuapp.com/user
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.895236: --> END GET
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.898686: --> GET https://mentorship-backend-temp.herokuapp.com/mentorship_relations/current
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.899332: --> END GET
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.911007: Has token!
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.916779: Has token!
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.934684: MembersPageShowed
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.945074: Transition { currentState: MembersPageInitial, event: MembersPageShowed, nextState: MembersPageLoading }
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.947598: --> GET https://mentorship-backend-temp.herokuapp.com/users/verified
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.947731: --> END GET
I/flutter ( 2865): INFO: 2020-05-03 03:16:13.951168: Has token!
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.896969: <-- 200 https://mentorship-backend-temp.herokuapp.com/mentorship_relations/current
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897167: via: 1.1 vegur
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897250: content-type: application/json
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897323: connection: keep-alive
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897392: date: Sun, 03 May 2020 01:16:15 GMT
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897461: content-length: 355
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897523: server: gunicorn/20.0.4
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897714: {
I/flutter ( 2865):     "id": 21,
I/flutter ( 2865):     "action_user_id": 157,
I/flutter ( 2865):     "sent_by_me": false,
I/flutter ( 2865):     "mentor": {
I/flutter ( 2865):         "id": 24,
I/flutter ( 2865):         "name": "Bartek"
I/flutter ( 2865):     },
I/flutter ( 2865):     "mentee": {
I/flutter ( 2865):         "id": 157,
I/flutter ( 2865):         "name": "ljco"
I/flutter ( 2865):     },
I/flutter ( 2865):     "creation_date": 1587751387.20845,
I/flutter ( 2865):     "accept_date": null,
I/flutter ( 2865):     "start_date": null,
I/flutter ( 2865):     "end_date": 1590278400.0,
I/flutter ( 2865):     "state": 2,
I/flutter ( 2865):     "notes": "test"
I/flutter ( 2865): }
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.897861: --> END GET (355-byte body)
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.907316: --> GET https://mentorship-backend-temp.herokuapp.com/mentorship_relation/21/tasks
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.909310: --> END GET
I/flutter ( 2865): INFO: 2020-05-03 03:16:14.915049: Has token!
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.148047: <-- 200 https://mentorship-backend-temp.herokuapp.com/user
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.148556: via: 1.1 vegur
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.148856: content-type: application/json
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.148949: connection: keep-alive
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.149019: date: Sun, 03 May 2020 01:16:15 GMT
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.149191: content-length: 785
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.149284: server: gunicorn/20.0.4
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.153850: {
I/flutter ( 2865):     "id": 24,
I/flutter ( 2865):     "name": "Bartek",
I/flutter ( 2865):     "username": "Pacia",
I/flutter ( 2865):     "email": "barpac02@gmail.com",
I/flutter ( 2865):     "password_hash": "pbkdf2:sha256:50000$PBgwD2Qi$1aed4b58ff957703b3c2cb919cc91ee99b9636a788a4fb67a1399e121b373aeb",
I/flutter ( 2865):     "terms_and_conditions_checked": true,
I/flutter ( 2865):     "is_admin": false,
I/flutter ( 2865):     "registration_date": 1584123761.49689,
I/flutter ( 2865):     "is_email_verified": true,
I/flutter ( 2865):     "email_verification_date": "2020-03-13T18:22:50.599463",
I/flutter ( 2865):     "bio": null,
I/flutter ( 2865):     "location": null,
I/flutter ( 2865):     "occupation": null,
I/flutter ( 2865):     "organization": null,
I/flutter ( 2865):     "slack_username": null,
I/flutter ( 2865):     "social_media_links": null,
I/flutter ( 2865):     "skills": null,
I/flutter ( 2865):     "interests": null,
I/flutter ( 2865):     "resume_url": null,
I/flutter ( 2865):     "photo_url": null,
I/flutter ( 2865):     "need_mentoring": true,
I/flutter ( 2865):     "available_to_mentor": true,
I/flutter ( 2865):     "current_mentorship_role": null,
I/flutter ( 2865):     "membership_status": null
I/flutter ( 2865): }
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.153996: --> END GET (785-byte body)
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.160559: Transition { currentState: ProfilePageLoading, event: ProfilePageShowed, nextState: ProfilePageSuccess }
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723152: <-- 200 https://mentorship-backend-temp.herokuapp.com/mentorship_relation/21/tasks
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723308: via: 1.1 vegur
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723373: content-type: application/json
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723494: connection: keep-alive
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723624: date: Sun, 03 May 2020 01:16:16 GMT
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723719: content-length: 165
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.723890: server: gunicorn/20.0.4
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.725271: [
I/flutter ( 2865):     {
I/flutter ( 2865):         "id": 1,
I/flutter ( 2865):         "description": "Hello! :D",
I/flutter ( 2865):         "is_done": false,
I/flutter ( 2865):         "created_at": 1588274567.920985,
I/flutter ( 2865):         "completed_at": null
I/flutter ( 2865):     }
I/flutter ( 2865): ]
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.725367: --> END GET (165-byte body)
I/flutter ( 2865): INFO: 2020-05-03 03:16:15.727953: Transition { currentState: RelationPageLoading, event: RelationPageShowed, nextState: RelationPageSuccess }
```

**Smartphone (please complete the following information):**
Pixel 3a emulator, API 29

PS I see `password hash` in the logs. Can it possible be used to retrieve my short, dummy password? 👀

**Comment by techno-disaster** [comment]
![Screenshot_20200503-221501535](https://user-images.githubusercontent.com/52817235/80920152-951e8f80-8d8b-11ea-8741-cbf5e5d5c415.jpg)
hmm, works fine on my end?

**Comment by bartekpacia** [comment]
Well, that's strange

**Comment by techno-disaster** [comment]
@isabelcosta can you test this? also can we tag the qa team to test such things?

**Comment by bartekpacia** [comment]
I tested it the on iPhone simulator, same result :/ (formatException blah blah).

**Comment by techno-disaster** [comment]
@bartekpacia i got this too. i also got the same error when i tried using curl and swagger to get the verified users list. Maybe it is a backend issue. 
cc: @isabelcosta

**Comment by bartekpacia** [comment]
indeed, it's more of a backend issue. We need to find out when it happens.

**Comment by techno-disaster** [comment]
@bartekpacia yep its a backend issue. The PR for it is already up [here](https://github.com/anitab-org/mentorship-backend/pull/541). we should be able to close it once its merged.

**Comment by techno-disaster** [comment]
We will have to change our side too. we have to specify page number in the request now

**Comment by isabelcosta** [comment]
@bartekpacia @Techno-Disaster it is a backend issue, sometimes the Heroku server responds badly. One of the reasons is that we don't have pagination implemented in the backend which means that everytime we call `GET /users` the entire database of users is queried and the entire list of users is returned, which is too much sometimes for the Heroku server

**Comment by techno-disaster** [comment]
> @bartekpacia @Techno-Disaster it is a backend issue, sometimes the Heroku server responds badly. One of the reasons is that we don't have pagination implemented in the backend which means that everytime we call `GET /users` the entire database of users is queried and the entire list of users is returned, which is too much sometimes for the Heroku server

so can we close this issue @isabelcosta ?

**Comment by isabelcosta** [comment]
Not sure, this is still an issue. If someone reports this again because they don't see this open, then we have duplicated issues @Techno-Disaster I think we can leave this open with a special label.

**Comment by isabelcosta** [comment]
@Techno-Disaster I removed available issue, because apparently there is nothing to do here, but "On Hold" because we are waiting on https://github.com/anitab-org/mentorship-backend/pull/541

**Comment by techno-disaster** [comment]
> Not sure, this is still an issue. If someone reports this again because they don't see this open, then we have duplicated issues @Techno-Disaster I think we can leave this open with a special label. 

makes sense . 👍

**Comment by techno-disaster** [comment]
Closing for now. 
Reason : Same as #85 and PR #87 is already up.
